rotkehlchen/chain/evm/tokens.py

```diff
@@ -201,6 +211,45 @@ def _query_chunks(
             total_token_balances = combine_dicts(total_token_balances, new_token_balances)
         return total_token_balances
 
+    def _compute_detected_tokens_info(self, addresses: list[ChecksumEvmAddress]) -> DetectedTokensType:  # noqa: E501
+        addresses_info: DetectedTokensType = defaultdict(lambda: (None, None))
+        # Need a mapping of proxy -> owner address
+        reversed_proxies_mappig = {v: k for k, v in self.proxies_mapping.items()}
+        with self.db.conn.read_ctx() as cursor:
+            for address in addresses:
+                # If `address` is a proxy, save token balances to its owner address
+                proxied_address = reversed_proxies_mappig.get(address, address)
+                new_tokens_list, new_last_queried_timestamp = self.db.get_tokens_for_address(
+                    cursor=cursor,
+                    address=address,
+                    blockchain=self.evm_inquirer.blockchain,
+                )
+                # Need these conditions in case we have a proxy and thus need to combine tokens
+                # of the proxy and the proxy's owner address
+                previous_tokens_list = addresses_info[proxied_address][0]
+                previous_last_queried_timestamp = addresses_info[proxied_address][1]
+                if previous_tokens_list is None:
+                    previous_tokens_list = new_tokens_list
```

Comment:
I'm not sure to understand what is happening here. Aren't you setting the list of tokens for the proxied address with all the tokens queried for the original address?

Reply:
Hmmm well, the logic of this function looks overcomplicated. I'll better try to find a cleaner and simpler approach.